lib/media/streaming_engine.js

```diff
@@ -3114,9 +3119,14 @@ shaka.media.StreamingEngine = class {
           'from=', mediaState.lastInitSegmentReference,
           'to=', reference.initSegmentReference);
 
+      const video = this.playerInterface_.video;
+      const pausedBeforeReset = video.paused;
       this.resetMediaSource(/* force= */ true).then(() => {
         const eventName = shaka.util.FakeEvent.EventName.BoundaryCrossed;
         this.playerInterface_.onEvent(new shaka.util.FakeEvent(eventName));
+        if (!pausedBeforeReset) {
+          video.play();
+        }
```

Comment:
`resetMediaSource` already do it... I think

Reply:
Not sure if this is the right approach but MSE might not recover (due to the gap situation I wrote above), and we'll loose the "previous autoplay" status. Playhead gets bumped over the gap (due to the timer, see previous comment) and MSE gets reset again. A play call would've been discarded as the previous reset did not properly handle autoplay.

Reply:
Reinitiating playback is now handled by MediaSource.